### DISCUSSION TITLE
How to plot the distribution histogram of geometrical compatibility factor m'?

### DISCUSSION THREAD

**Discussion by 1996phd** [mtex-toolbox/mtex#1407] [Ask Anything] [answered]
Hello, everyone!
Now I have obtained the m' value of each grain boundary through the 'mPrime' function. Next I'm interested in the statistical distribution of m' value, how could I get that? I mean, a histogram to show the frequency of m' from 0-1.

**Comment by 1996phd** [comment]
By the way, I hope that the frequency is related to the length of grain boundary, that is, a longer grain boundary will exhibit a larger frequency of its m' value

**Comment by kilir** [comment, marked as answer]
Hi,
Matlab provides functions such as histogram. In case you are interested in a statistical distribution of values in the sense of a pdf, you can use the Matlab calcDensity. You can derive a weight for your histogram/density estimate from e.g. grains.boundary.segLength in case your grains are smoothed.

Cheers,
Rüdiger

**Reply by 1996phd** [reply]
Thanks!
But I still don't know how the calcDensity function is used in this problem. Instead, a rather simple but not smart method is used to draw that figure...
`gBlength=gB.segLength
n=length(mP);
count=zeros(10,1);
for i=1:n
    if mP(i)<=0.1
        count(1)=count(1)+gBlength(i);
    else if mP(i)>0.1&&mP(i)<=0.2
            count(2)=count(2)+gBlength(i);
        else if mP(i)>0.2&&mP(i)<=0.3
                count(3)=count(3)+gBlength(i);
            else if mP(i)>0.3&&mP(i)<=0.4
                    count(4)=count(4)+gBlength(i);
                else if mP(i)>0.4&&mP(i)<=0.5
                        count(5)=count(5)+gBlength(i);
                    else if mP(i)>0.5&&mP(i)<=0.6
                            count(6)=count(6)+gBlength(i);
                        else if mP(i)>0.6&&mP(i)<=0.7
                                count(7)=count(7)+gBlength(i);
                            else if mP(i)>0.7&&mP(i)<=0.8
                                    count(8)=count(8)+gBlength(i);
                                else if mP(i)>0.8&&mP(i)<=0.9
                                        count(9)=count(9)+gBlength(i);
                                    else if mP(i)>0.9&&mP(i)<=1.0
                                            count(10)=count(10)+gBlength(i);
                                        end
                                    end
                                end
                            end
                        end
                    end
                end
            end
        end
    end
end
figure(2)
count=count/sum(count);
x=0.05:0.1:0.95;
bar(x,count)`

**Reply by 1996phd** [reply]
And simplified as follows:
`gBlength=gB.segLength`
`n=length(mP);`
`nn=10;`
`count=zeros(nn,1);`
`for i=1:n`
`    if mP(i)<=1/nn`
`        count(1)=count(1)+gBlength(i);`
`    end`
`    for j=2:nn`
`        if  mP(i)>(j-1)/nn&&mP(i)<=j/nn`
`            count(j)=count(j)+gBlength(i);`
`        end`
`    end`
`end`
`figure(2)`
`count=count/sum(count);`
`x=0.05:0.1:0.95;`
`bar(x,count)`
